### PR TITLE
Move git.DiffPath to gitserver.Client

### PR DESCRIPTION
**Pull request by vrto**
### Description

This change is a part of a larger effort to reorganize code in gitserver. The Repository Management Team is moving git commands from `internal/vcs/git` to `internal/gitserver`.

See this issue for more details https://github.com/sourcegraph/sourcegraph/issues/33280

Closes https://github.com/sourcegraph/sourcegraph/issues/33439

## Test plan

Code only moved around, all tests still pass.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/codeintel/resolvers/position.go [hunk 1]
```diff
@@ -35,15 +35,16 @@ type PositionAdjuster interface {
 }
 
 type positionAdjuster struct {
-	db        database.DB
+	client    gitserver.Client
```
Comment (vrto) [inline enterprise/cmd/frontend/internal/codeintel/resolvers/position.go:38]: @sashaostrikov I think you meant to do this sooner or later, so I took the initiative as it was required for this work :)

File: internal/vcs/git/exec.go [hunk 2]
```diff
@@ -48,28 +47,6 @@ func execSafe(ctx context.Context, db database.DB, repo api.RepoName, params []s
 	return stdout, stderr, exitCode, err
 }
 
-// execReader executes an arbitrary `git` command (`git [args...]`) and returns a
-// reader connected to its stdout.
-//
-// execReader should NOT be exported. We want to limit direct git calls to this
-// package.
-func execReader(ctx context.Context, db database.DB, repo api.RepoName, args []string) (io.ReadCloser, error) {
-	if Mocks.ExecReader != nil {
```
Comment (ryanslade) [inline internal/vcs/git/exec.go:57]: I think we can safely remove this Mock field now.

**Review by ryanslade — APPROVED** [review]

File: enterprise/cmd/frontend/internal/codeintel/resolvers/position.go (hunk 1 above)
Comment (sashaostrikov) [reply enterprise/cmd/frontend/internal/codeintel/resolvers/position.go:38]: thanks, Michal! :)

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: internal/vcs/git/exec.go (hunk 2 above)
Comment (vrto) [reply internal/vcs/git/exec.go:57]: Still needed in a few places!
Screenshot 2022-04-07 at 9 08 00

**Pull request merged**